services/server/src/server/controllers/repository/repository.handlers.ts

```diff
@@ -128,9 +136,71 @@ export async function checkAllByChainAndAddressEndpoint(
             });
           }
 
-          map
-            .get(address)
-            .chainIds.push({ chainId, status: found[0].runtimeMatch });
+          // Proxy detection and resolution
+          type Implementation = { address: string; name?: string };
+          let proxyStatus: {
+            isProxy?: boolean;
+            proxyType?: ProxyType | null;
+            implementations?: Implementation[];
+            proxyResolutionError?: string;
+          } = {};
+          if (
+            req.query.resolveProxies === "true" &&
+            found[0].onchainRuntimeBytecode
```

Comment:
Just to make sure, here we use the `onchainRuntimeBytecode` from the Database right? Not fresh from an RPC

Reply:
yes exactly. That's why I added it to the database querying method.